v4.3.1.cw1

fteem opened the pull request:
~Not to be merged. We need to review the code here, and open a PR upstream. If the maintainers accept the changes they'll need to bump the version and cut a new gem.~

Since we don't see any development upstream on our PR (but there's a `v5` branch with ongoing development), we need to resort to our own fork of the gem for the time being. When the upstream maintainers get out a new version of the gem, without the current pitfalls/flaws, we can move to the upstream version.

This version of our fork will use the same version as the upstream gem, but we will apply `cw1` as a "teeny" version to it. After merging this PR, I will push a new tag to be used in `cw-shipping`.

Upstream PRs:
- Removal of retry mechanism PR - https://github.com/AfterShip/aftership-sdk-ruby/pull/23

fteem committed:
Update README

fteem committed:
Add AfterShip::V4::Configuration

fteem committed:
Remove retry mechanism

fteem committed:
Set send_timeout to HTTPClient instance from AfterShip.configuration

fteem committed:
Inject Aftership.configuration.headers to HTTPClient requests headers

fteem committed:
Rescue HTTPClient::SendTimeoutError if HTTPClient configuration triggers the error

fteem committed:
Modify AfterShip::V4::Base#url to use API endpoint via AfterShip.configuration.api_endpoint

fteem committed:
Add AfterShip.configuration

coalwater commented on README.md:
```diff
@@ -9,6 +9,11 @@ This extension helps developers to integrate with AfterShip easily.
 
 ### Changes
+* 2017-12-06 4.4.0
+  - Introduced `Configuration` object that encapsulates any gem configuration, like client `timeout` (in seconds), `headers`, `api_key` and `api_endpoint` (that defaults to `api.aftership.com`)
```
Actually the timeout is in milliseconds, because you divide by 1000

fteem committed:
Add more granularity to client timeouts configurations

fteem committed:
Use the granular timeouts to configure HTTPClient

fteem committed:
Handle HTTPClient::ConnectTimeoutError when issuing a call

fteem committed:
Handle HTTPClient::ReceiveTimeoutError when issuing a call

Commit:
Set cf_ray variable from the response headers

Commit:
Remove existence check while checking timeout config

There is a default value in the configuration object, so this will
always exist

Commit:
Update rspec to 3.7

Commit:
Add a spec for AfterShip::V4::Base
